web/impact/impact/tests/api_test_case.py

```diff
@@ -98,5 +98,10 @@ def assert_options_include(self, method, expected_options, object_id=None):
                 self.assertIn(key, options)
                 self.assertEqual(options[key], params)
 
-    def assert_ui_notification(self, response, notification):
-        self.assertEqual(response.data, notification)
+    def assert_ui_notification(self, response, success, notification):
+        expected_ui_notification = {
+            'success': success,
+            'header': self.success_header if success else self.fail_header,
+            'detail': notification
+        }
+        self.assertEqual(response.data, expected_ui_notification)
```

Comment:
Is there any possibility that response.data might ever include information that is not part of the notification bundle? For example, a list of created objects, or something of that sort?

If that is a possibility, then we should check the individual keys, something like:
```
    def assert_ui_notification(self, response, success, notification):
        data = response.data
        self.assertTrue(all(
            data['success'] == success,
            data['header'] == self.success_header if success else self.fail_header,
            data['detail'] == notification
        ), msg="Notification data was not as expected")

```

It would be possible to make this more informative, by including the precise failure in the `msg` but we can save that for another day.

Reply:
from the current office hours tickets, the response for create/update doesn't return any additional data however I think asserting for the individual keys is the best idea in the long run. I'm on it